datafusion/physical-optimizer/src/pruning.rs

```diff
@@ -819,16 +813,24 @@ impl RequiredColumns {
     /// statistics column, while keeping track that a reference to the statistics
     /// column is required
     ///
-    /// for example, an expression like `col("foo") > 5`, when called
+    /// For example, an expression like `col("foo") > 5`, when called
     /// with Max would result in an expression like `col("foo_max") >
-    /// 5` with the appropriate entry noted in self.columns
+    /// 5` with the appropriate entry noted in self.columns.
+    ///
+    /// Along with a reference to the min/max/null_count/row_count column
+    /// an expression for `{stats_col} IS NULL` is returned that should be `OR`ed
+    /// with any expressions applied to the column itself such that if any value in `{stats_col}`
+    /// is `null` we return `true` and not `null` because `null` is falsy, but we'd need to scan that
+    /// container, causing confusion and preventing optimizations because of the inverted logic!
+    ///
+    /// Thus for the example above we'd arrive at the final expression `foo_max is null or foo_max > 5`.
     fn stat_column_expr(
         &mut self,
         column: &phys_expr::Column,
         column_expr: &Arc<dyn PhysicalExpr>,
         field: &Field,
         stat_type: StatisticsType,
-    ) -> Result<Arc<dyn PhysicalExpr>> {
+    ) -> Result<(Arc<dyn PhysicalExpr>, Arc<dyn PhysicalExpr>)> {
```

Comment:
The idea here is to force callers to do something with the null check expression, or explcitly discard it.
This let me update all call sites with confidence of not having missed one, in addition to nicely abstracting away the logic.